File: Ems_frontend/src/services/EmployeeService.js
import apiClient from "../apiConfig";

const REST_API_BASE_URL = "/employees";

export const listEmployees = (query = '', departmentId = '', teamId = '') => {
  const params = new URLSearchParams();
  if (query) params.append('query', query);
  if (departmentId) params.append('departmentId', departmentId);
  if (teamId) params.append('teamId', teamId);
  params.append('size', '100'); // Default to 100 to show more employees
  return apiClient.get(REST_API_BASE_URL + '?' + params.toString());
};

export const getEmployeesByDesignation = (designation) => apiClient.get(`${REST_API_BASE_URL}/designation/${designation}`);

export const getEmployeesBySkill = (skillName) => apiClient.get(`${REST_API_BASE_URL}/skill/${skillName}`);

export const addSkillToEmployee = (employeeId, skillName) => apiClient.post(`${REST_API_BASE_URL}/${employeeId}/skills?skillName=${skillName}`);


export const createEmployee = (employee) => apiClient.post(REST_API_BASE_URL, employee);

export const getEmployee = (employeeId) => apiClient.get(REST_API_BASE_URL + '/' + employeeId);

export const updateEmployee = (employeeId, employee) => apiClient.put(REST_API_BASE_URL + '/' + employeeId, employee);
  
export const deleteEmployee = (employeeId) => apiClient.delete(REST_API_BASE_URL + '/' + employeeId);

export const getLoggedInEmployee = () => apiClient.get(REST_API_BASE_URL + '/me');

File: Ems_backend/src/main/java/com/ak/ems/service/EmployeeService.java
package com.ak.ems.service;

import com.ak.ems.dto.EmployeeDto;
import com.ak.ems.response.PageResponse;
import java.util.List;




public interface EmployeeService {
    EmployeeDto createEmployee(EmployeeDto employeeDto);

    EmployeeDto getEmployeeById(Long employeeId);

    PageResponse<EmployeeDto> getAllEmployees(String query, Long departmentId, Long teamId, int page, int size, String sortBy, String sortDir);

    EmployeeDto updateEmployee(Long employeeId, EmployeeDto updatedEmployeeDto);

    EmployeeDto getEmployeeByUsername(String username);

    List<EmployeeDto> getEmployeesByDesignation(String designation);

    List<EmployeeDto> getEmployeesBySkill(String skillName);

    void addSkillToEmployee(Long employeeId, String skillName);

    void deleteEmployee(Long employeeId);
}
